crates/wallet/src/wallet/mod.rs

```diff
@@ -1457,8 +1458,8 @@ impl Wallet {
                     .index
                     .next_unused_spk(change_keychain)
```

Comment:
It's a bug to reveal something when we error out. Calling `.next_unused_spk` may reveal an spk here. I think it makes sense to fix this in the same PR.